### ISSUE TITLE
What were the major learnings from `Inquirer.js` ?

### ISSUE THREAD

**Issue by hemanth** [SBoudrias/ama#10]

**Comment by SBoudrias** [comment]
Well Inquirer is my first major open source project (where I'm the main author). So lots of what I discovered is about open source:
1. The community will back good ideas. I didn't had a huge audience when launching Inquirer.js, but it became a fairly big project on its own.
2. People will rely on you and most of them won't give much help. Users will open a bunch of tickets, but very few will actually send PR.
3. People like amazing UI, and they'll port it to other language... Inquirer in python, in ruby, etc. I really never expected this.

About Inquirer of itself.

I think it's very hard to create a reliable product without integration test. This is still an issue today, but I really dream of a command line integration testing tool.

Command lines are very inconsistent between each other. Each one have tons of features other don't support, so it's pretty hard to find the minimum common features and rely on them.

At the time Node had a very bad support for ANSI control codes, so I created Readline2 to have a common base to work off.

Otherwise from my recent refactor, I'd say my design ideal today is: very few classes, lots of standalone functions and englobe every common pattern in a reusable components who're easier to test.
